### ISSUE TITLE
Updating symfony validation to 2.7.0 brings deprecated warnings

### ISSUE THREAD

**Issue by Ronnie-J** [silexphp/Silex#1170]
Hi guys.

Ive just updated my composer dependencies and symfony/validation went from;
```
  - Removing symfony/validator (v2.6.7)
  - Installing symfony/validator (v2.7.0)
 ```
After that I started getting deprecated warnings:

```
PHP Deprecated:  The Symfony\Component\Validator\Validator class is deprecated since version 2.5 and will be removed in 3.0. Use the Symfony\Component\Validator\Validator\RecursiveValidator class instead. in /vagrantenvironments/phpcsa/application/vendor/symfony/validator/Validator.php on line 14

PHP Deprecated: The Symfony\Component\Validator\ConstraintValidator::buildViolation is deprecated since version 2.5 and will be removed in 3.0. in /vagrantenvironments/phpcsa/application/vendor/symfony/validator/ConstraintValidator.php on line 68
```

I suppose the ValidationServiceProvider needs some updating?

**Comment by fabpot** [comment]
Which version of Silex are you using? 1.2 is not "compatible" with Symfony 2.7, but the upcoming Silex 1.3 is.

**Comment by Ronnie-J** [comment]
Im on 1.2 - yes I kinda expected that :) 

Its still an issue though and 1.3 would be very much appreciated. Releasing Symfony2 2.7 is a major step but a lot of us is dependent on Silex as well.

**Comment by davedevelopment** [comment]
The 1.3 branch should be fine, I think those problems will be caused by "userland" code. I upgraded a Silex 1.3 app to the 2.7 components the other day and had to deal with a bunch of these, but they were all in my own code.

**Comment by Ronnie-J** [comment]
I dont see any code that by my self should trigger this dependency warning. Im pretty sure its the ValidationProvider that needs updating after changes made to symfony/validation (2.7).

Im expecting that would be fixed in 1.3 when released :) :+1:

**Comment by stof** [comment]
2.7.1 does not show these warnings by default anymore no (it requires using a special error handler catching them even when silenced, like the Symfony Debug ErrorHandler which logs them)
